### DISCUSSION TITLE
search by filename

### DISCUSSION THREAD

**Discussion by gsgoldma** [civitai/civitai#183] [Ideas]
Some of the filenames don't match up. S,o if you want to look, at let's say the keyword for your model, you have to remember the name on the model on the page. But you might not remember that and only have the filename to go off of.

**Comment by CarbonatedPastries** [comment]
It would also be helpful to be able to search by Hashes and SHA256s. I tested a LoRA I got off the site using a prompt from one of the example images and got something not only different from the LoRA but even my base model. I wanted to figure out what model they used, but the prompt information only has the model hash, which both the site and Google returned no results.

**Comment by JustMaier** [comment]
We might have shot ourselves in the foot on this one. Currently we generate the filenames on demand using some logic and certain information about the model file that you're downloading. This can change from time to time, so if you downloaded a file last month for example and you downloaded that same file this month, they'll likely have slightly different names because of tuning we've done around our file naming logic.

That said, you should be able to search by hash. We currently hash with the following algorithms:
- SHA256
- AutoV1 (first 8 characters of the SHA256 hash of the first 100MB)
- AutoV2 (first 10 characters of the full SHA256)
- CRC32
- Blake3

Hashing on upload was added at the beginning of this month, and we just recently added the last 3 hash options on that list. We're working on sending everything back through the scanner, but it might be a week or two before that's done.

**Edit:** Everything has been sent through the scanner and now has these hashes! 🎉 🎉 🎉

**Reply by gsgoldma** [reply]
maybe going forward there could be something you could add to the filename that would make it easily searchable.
text embeddings dont have hashes and they have strange names sometimes e.g., rdf55, so it can be hard to remember what the heck it even is....even after like 5 minutes let alone days if you downloaded more than one thing, especially since that file name currently doesn't show up on your download history or in the page itself.

**Reply by CarbonatedPastries** [reply]
Adding hashes to other upload types such as embeddings probably isn't too much effort, and if you have those, you don't need filenames. Especially important in case someone decides to rename their files to make them easier and more organized to reference in their preferred frontend.

Searching by multiple tags at once would be interesting, too. And of course by description and version text. But that is a lot more to index.

**Reply by JustMaier** [reply]
@gsgoldma we recently added the ability to search for things by their trained word, so that should make it possible to find an embedding just using the name of it in your embedding folder.

**Reply by JustMaier** [reply]
@CarbonatedPastries 
> Adding hashes to other upload types such as embeddings probably isn't too much effort

We've added hashes to embeddings and other resources as well. Now there just needs to be an easier way for individuals to get the hashes of their embeddings 😛 

Regarding the additional search options and fields, yeah, our search/filter options are fairly weak right now. It's a constraint of our current infrastructure which is constrained by our resources. Hopefully we can improve this in the future :)

**Comment by Weixuanf** [comment]
+1 this one. I'm building a ComfyUI automatic model installer, it would be reaaaally useful to be able to search by the downloaded name of the model. For example, the downloaded file name could be MoXinV1.safetensor, but if I use API endpoint to search using query=MoXinV1, I will get nothing, because the model name is Mo Xin

**Comment by Weixuanf** [comment]
any update? really hope to get this!
